woocommerce/woocommerce
# Decode HTML entities and strip HTML tags in product names for cart quantity change notifications

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #50540.



### How to test the changes in this Pull Request:



1. Create a product with a name that includes HTML tags (such as `b`) and characters that will be encoded as HTML entities, such as:
    - `-` (`&#8211;`)
    - `"` (`&#8243;`)
2. Add product to cart
3. Open cart in 2nd browser window
4. Remove product from cart in 2nd browser window
5. In first browser window, increase product quantity in cart
6. Observe that notification shown strips HTML tags and decodes HTML entities

<img width="1276" alt="Screenshot 2024-08-09 at 13 59 16" src="https://github.com/user-attachments/assets/f984014d-706a-4e0c-9b35-914ff1bf5e27">



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
